src/flashcart/ed64/ed64.c

```diff
@@ -15,45 +15,79 @@
 
 // #include "../menu/settings.h"
 
-// // This is a trial hack before using the settings API.
-// #ifndef LAST_ROM_FILE_PATH
-// #define LAST_ROM_FILE_PATH "/ed64/last_rom.txt"
-// #endif
-
+// This is a trial hack before using the settings API.
+#ifndef LAST_SAVE_FILE_PATH
+#define LAST_SAVE_FILE_PATH strip_sd_prefix("/menu/last_rom.txt")
+#endif
 
 extern int ed_exit(void);
 
 static flashcart_err_t ed64_init (void) {
 
-    // TODO: partly already done, see https://github.com/DragonMinded/libdragon/blob/4ec469d26b6dc4e308caf3d5b86c2b340b708bbd/src/libcart/cart.c#L1064
-
-    // FIXME: Update firmware if needed.
-    // FIXME: Enable RTC if available.
+    if (file_exists(strip_sd_prefix("/menu/RESET"))) { // checks if reset
+    
+    f_unlink(strip_sd_prefix("/menu/RESET"));
+        
+    FIL lrp_fil;
+    UINT lrp_br;
+    TCHAR lrp_path[255*32];
 
-    // FIXME: Retrive a config file from the SD card that might have been set.
-    // This should include the location of the ROM and its save type.
-    // Then, if it is valid, perform a save.
-    // FIL lrp_fil;
-    // UINT lrp_br;
+    // FIXME: use max supported length
+    
+    if (f_open(&lrp_fil, LAST_SAVE_FILE_PATH, FA_READ) != FR_OK) {
+        return FLASHCART_ERR_LOAD;
+    }
 
-    // if (f_open(&lrp_fil, LAST_ROM_PATH_FILE, FA_READ) != FR_OK) {
-    //     return FLASHCART_ERR_LOAD;
-    // }
+    int lrp_size = f_size(&lrp_fil);
 
-    // if (f_read(&lrp_fil, lrp_path, 1024, &lrp_br) != FR_OK) {
-    //     f_close(&lrp_fil);
-    //     return FLASHCART_ERR_LOAD;
-    // }
+    if (f_read(&lrp_fil, lrp_path, lrp_size, &lrp_br) != FR_OK) {
```

Comment:
Given we know the lrp file size, we should really be allocating the size of TCHAR to that (so move the declaration). It will need to be one CHAR larger as it needs to take into account the null CHAR.